File: androidApp/features/profile/src/main/kotlin/com/cineverse/android/features/profile/OnboardingScreen.kt
package com.cineverse.android.features.profile

import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cineverse.core.designsystem.components.CineVerseButton
import kotlinx.coroutines.launch

data class OnboardingPage(
    val title: String,
    val description: String,
    val icon: ImageVector,
    val iconColor: Color
)

@OptIn(ExperimentalFoundationApi::class)
@Composable
fun OnboardingScreen(
    onComplete: () -> Unit,
    modifier: Modifier = Modifier
) {
    val pages = listOf(
        OnboardingPage(
            title = "Welcome to CineVerse",
            description = "Discover your next favorite movie, track releases, and build your ultimate watchlist in one clean application.",
            icon = Icons.Default.PlayArrow,
            iconColor = Color(0xFFE11D48) // Rose
        ),
        OnboardingPage(
            title = "Dynamic Movie Roll",
            description = "Can't decide what to watch? Roll the interactive 3D dice to randomly select a top-rated movie from our database.",
            icon = Icons.Default.Favorite,
            iconColor = Color(0xFF3B82F6) // Blue
        ),
        OnboardingPage(
            title = "Smart Watchlist Reminders",
            description = "Add movie releases to your watchlist and schedule custom notifications so you never miss a premiere.",
            icon = Icons.Default.Notifications,
            iconColor = Color(0xFF10B981) // Green
        )
    )

    val pagerState = rememberPagerState(pageCount = { pages.size })
    val coroutineScope = rememberCoroutineScope()

    Column(
        modifier = modifier
            .fillMaxSize()
            .background(Color(0xFF0F172A)) // Slate Background
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        // Skip Button
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.End
        ) {
            if (pagerState.currentPage < pages.size - 1) {
                TextButton(
                    onClick = onComplete
                ) {
                    Text(
                        text = "Skip",
                        color = Color(0xFF94A3B8),
                        fontWeight = FontWeight.Medium
                    )
                }
            } else {
                Spacer(modifier = Modifier.height(48.dp))
            }
        }

        HorizontalPager(
            state = pagerState,
            modifier = Modifier
                .weight(1f)
                .fillMaxWidth()
        ) { pageIndex ->
            val page = pages[pageIndex]
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(horizontal = 16.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.Center
            ) {
                Box(
                    modifier = Modifier
                        .size(150.dp)
                        .clip(CircleShape)
                        .background(page.iconColor.copy(alpha = 0.1f)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = page.icon,
                        contentDescription = null,
                        modifier = Modifier.size(80.dp),
                        tint = page.iconColor
                    )
                }

                Spacer(modifier = Modifier.height(40.dp))

                Text(
                    text = page.title,
                    color = Color.White,
                    style = MaterialTheme.typography.headlineLarge,
                    fontWeight = FontWeight.Bold,
                    textAlign = TextAlign.Center
                )

                Spacer(modifier = Modifier.height(16.dp))

                Text(
                    text = page.description,
                    color = Color(0xFF94A3B8), // Text Secondary Slate
                    fontSize = 16.sp,
                    textAlign = TextAlign.Center,
                    lineHeight = 24.sp
                )
            }
        }

        // Indicators & Buttons
        Column(
            modifier = Modifier.fillMaxWidth(),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // Page Indicator Dots
            Row(
                horizontalArrangement = Arrangement.spacedBy(8.dp),
                modifier = Modifier.padding(bottom = 32.dp)
            ) {
                repeat(pages.size) { index ->
                    val isSelected = pagerState.currentPage == index
                    Box(
                        modifier = Modifier
                            .size(if (isSelected) 24.dp else 8.dp, 8.dp)
                            .clip(CircleShape)
                            .background(
                                if (isSelected) Color(0xFFE11D48) else Color(0xFF334155)
                            )
                    )
                }
            }

            // Action Button
            val isLastPage = pagerState.currentPage == pages.size - 1
            CineVerseButton(
                text = if (isLastPage) "Get Started" else "Next",
                onClick = {
                    if (isLastPage) {
                        onComplete()
                    } else {
                        coroutineScope.launch {
                            pagerState.animateScrollToPage(pagerState.currentPage + 1)
                        }
                    }
                },
                modifier = Modifier.fillMaxWidth()
            )
        }
    }
}

File: iosApp/Screens/OnboardingScreen.swift
import SwiftUI
import shared

struct OnboardingSlide {
    let title: String
    let description: String
    let iconName: String
    let iconColor: Color
}

struct OnboardingScreen: View {
    let onComplete: () -> Void
    
    @State private var currentPage = 0
    
    private let slides = [
        OnboardingSlide(
            title: "Welcome to CineVerse",
            description: "Discover your next favorite movie, track releases, and build your ultimate watchlist in one clean application.",
            iconName: "film",
            iconColor: AppColors.primary
        ),
        OnboardingSlide(
            title: "Dynamic Movie Roll",
            description: "Can't decide what to watch? Roll the interactive 3D dice to randomly select a top-rated movie from our database.",
            iconName: "dice.fill",
            iconColor: .blue
        ),
        OnboardingSlide(
            title: "Smart Watchlist Reminders",
            description: "Add movie releases to your watchlist and schedule custom notifications so you never miss a premiere.",
            iconName: "bell.fill.badge.play.fill",
            iconColor: .green
        )
    ]
    
    var body: some View {
        VStack(spacing: 20) {
            
            HStack {
                Spacer()
                if currentPage < slides.count - 1 {
                    Button(action: onComplete) {
                        Text("Skip")
                            .font(.system(size: 16, weight: .medium))
                            .foregroundColor(AppColors.textSecondary)
                    }
                    .padding(.trailing, 24)
                    .padding(.top, 16)
                } else {
                    Spacer().frame(height: 40)
                }
            }

            TabView(selection: $currentPage) {
                ForEach(0..<slides.count, id: \.self) { index in
                    let slide = slides[index]
                    VStack(spacing: 30) {
                        ZStack {
                            Circle()
                                .fill(slide.iconColor.opacity(0.1))
                                .frame(width: 150, height: 150)
                            
                            Image(systemName: slide.iconName)
                                .resizable()
                                .aspectRatio(contentMode: .fit)
                                .frame(width: 80, height: 80)
                                .foregroundColor(slide.iconColor)
                        }
                        
                        Text(slide.title)
                            .font(AppTypography.headingLarge)
                            .foregroundColor(AppColors.textPrimary)
                            .multilineTextAlignment(.center)
                            .padding(.horizontal, 24)
                        
                        Text(slide.description)
                            .font(AppTypography.bodyMedium)
                            .foregroundColor(AppColors.textSecondary)
                            .multilineTextAlignment(.center)
                            .lineSpacing(4)
                            .padding(.horizontal, 32)
                    }
                    .tag(index)
                }
            }
            .tabViewStyle(PageTabViewStyle(indexDisplayMode: .never))

            HStack(spacing: 8) {
                ForEach(0..<slides.count, id: \.self) { index in
                    Capsule()
                        .fill(currentPage == index ? AppColors.primary : AppColors.textSecondary.opacity(0.4))
                        .frame(width: currentPage == index ? 24 : 8, height: 8)
                }
            }
            .padding(.bottom, 24)

            let isLastPage = currentPage == slides.count - 1
            CineVerseButton(
                text: isLastPage ? "Get Started" : "Next",
                onClick: {
                    if isLastPage {
                        onComplete()
                    } else {
                        withAnimation {
                            currentPage += 1
                        }
                    }
                }
            )
            .padding(.horizontal, 24)
            .padding(.bottom, 32)
        }
        .background(AppColors.background.edgesIgnoringSafeArea(.all))
    }
}
